src/core_plugins/timelion/public/directives/timelion_expression_input_helpers.js

```diff
@@ -85,36 +85,49 @@ async function extractSuggestionsFromParsedResult(result, cursorPosition, functi
   const functionHelp = functionList.find((func) => {
     return func.name === activeFunc.function;
   });
-  const providedArguments = activeFunc.arguments.map((arg) => {
-    return arg.name;
-  });
 
-  // return function suggestion if cursor is outside of parentheses
+  // return function suggestion when cursor is outside of parentheses
   // location range includes '.', function name, and '('.
   const openParen = activeFunc.location.min + activeFunc.function.length + 2;
   if (cursorPosition < openParen) {
     return { list: [functionHelp], location: activeFunc.location, type: SUGGESTION_TYPE.FUNCTIONS };
   }
 
-  // Do not provide 'inputSeries' as argument suggestion for chainable functions
-  const args = functionHelp.chainable ? functionHelp.args.slice(1) : functionHelp.args.slice(0);
-
+  // return argument value suggestions when cursor is inside agrument value
   const activeArg = activeFunc.arguments.find((argument) => {
     return inLocation(cursorPosition, argument.location);
   });
-  // return argument_value suggestions when cursor is inside agrument value
   if (activeArg && activeArg.type === 'namedArg' && inLocation(cursorPosition, activeArg.value.location)) {
-    const valueSuggestions = await getArgValueSuggestions(
-      activeArg.name,
-      activeArg.value.text,
-      functionHelp.args.find((arg) => {
+    const {
```

Comment:
It might be nice to destructure the methods in `argValueSuggestions`, too.

```js
const {
  hasDynamicSuggestionsForArgument,
  getDynamicSuggestionsForArgument,
  getStaticSuggestionsForInput,
} = argValueSuggestions;
```

This could make the code on lines 112 through 120 a bit terser.